src/main/java/games/strategy/triplea/oddsCalculator/ta/OddsCalculator.java

```diff
@@ -52,205 +59,210 @@
-    final List<Unit> defenderOrderOfLosses =
-        OddsCalculator.getUnitListByOrderOfLoss(this.defenderOrderOfLosses, defendingUnits, gameData);
-    for (int i = 0; i < count && !cancelled; i++) {
-      final CompositeChange allChanges = new CompositeChange();
-      final DummyDelegateBridge bridge1 =
-          new DummyDelegateBridge(attacker, gameData, allChanges, attackerOrderOfLosses, defenderOrderOfLosses,
-              keepOneAttackingLandUnit, retreatAfterRound, retreatAfterXUnitsLeft, retreatWhenOnlyAirLeft);
-      final GameDelegateBridge bridge = new GameDelegateBridge(bridge1);
-      final MustFightBattle battle = new MustFightBattle(location, attacker, gameData, battleTracker);
-      battle.setHeadless(true);
-      battle.isAmphibious();
-      battle.setUnits(defendingUnits, attackingUnits, bombardingUnits,
-          (amphibious ? attackingUnits : new ArrayList<>()), defender, territoryEffects);
-      bridge1.setBattle(battle);
-      battle.fight(bridge);
-      aggregateResults.addResult(new BattleResults(battle, gameData));
-      // restore the game to its original state
-      gameData.performChange(allChanges.invert());
-      battleTracker.clear();
-      battleTracker.clearBattleRecords();
-    }
+
+    // ensure Configuration reference is published correctly between threads
+    final Configuration configuration = this.configuration;
+    final AggregateResults aggregateResults = IntStream.range(0, configuration.runCount).parallel()
+        .filter(it -> !cancelled)
+        .mapToObj(it -> {
+          try {
+            final SimulationContext context = contextQueue.take();
+            try {
+              return Optional.of(runSimulation(configuration, context));
+            } finally {
+              contextQueue.add(context);
+            }
+          } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return Optional.<BattleResults>empty();
+          }
+        })
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .collect(AggregateResults.unionCollector(configuration.runCount));
     aggregateResults.setTime(System.currentTimeMillis() - start);
     isRunning = false;
     cancelled = false;
     return aggregateResults;
   }
 
-  @Override
-  public AggregateResults call() {
-    return calculate();
+  // this method may be run concurrently
+  private static BattleResults runSimulation(final Configuration configuration, final SimulationContext context) {
+    final CompositeChange allChanges = new CompositeChange();
+    try {
+      context.gameData
+          .performChange(ChangeFactory.removeUnits(context.location, context.location.getUnits().getUnits()));
+      context.gameData.performChange(ChangeFactory.addUnits(context.location, context.attackingUnits));
+      context.gameData.performChange(ChangeFactory.addUnits(context.location, context.defendingUnits));
```

Comment:
The undos are because the `GameData`, even though it is a clone of the original, is reused by multiple parallel tasks (e.g. on a two-core machine, we make two clones, but those two clones may be used to run 10,000 simulations).  Presumably, each task needs a clean slate at the start of each run.  I didn't verify if this is _not_ the case because it was existing code.